client_apis/python/src/tests/integration/watchlist_test.py

```diff
@@ -0,0 +1,420 @@
+
+        add_action_result = cb.watchlist_action_add(
+            watchlist_id=watchlist_id,
+            action_type='email',
+            email_recipient_user_ids=[1]
+        )
+        self.assertGreater(add_action_result["action_id"], 0)
+
+        actions = cb.watchlist_action_get(watchlist_id=watchlist_id)
+        self.assertIsNotNone(actions)
+        self.assertEqual(len(actions), 1)
+        self.assertEqual(actions[0]["id"], add_action_result["action_id"])
+        self.assertEqual(actions[0]["action_data"], "{\"email_recipients\":[1]}")
+
+        modify_result = cb.watchlist_action_del(
+            watchlist_id=watchlist_id,
+            action_id=add_action_result["action_id"]
+        )
+        self.assertEqual(modify_result["result"], "success")
+
+        new_actions = cb.watchlist_action_get(watchlist_id=watchlist_id)
+        self.assertIsNotNone(new_actions)
+        self.assertEqual(len(new_actions), 0)
+        return
+
+    def test_delete_invalid_watchlist_action(self):
+        add_result = cb.watchlist_add(
+            type="modules",
+            name="%s %s" % (name_prefix, datetime.now().isoformat(' ')),
+            search_query="q=is_executable_image%3Afalse&cb.urlver=1&sort=server_added_timestamp%20desc",
+            readonly=False
+        )
+        watchlist_id = add_result["id"]
+        self.assertGreater(add_result["id"], 0)
+
+        actions = cb.watchlist_action_get(watchlist_id=watchlist_id)
+        self.assertIsNotNone(actions)
+        self.assertEqual(len(actions), 0)
+
+        add_action_result = cb.watchlist_action_add(
+            watchlist_id=watchlist_id,
+            action_type='email',
+            email_recipient_user_ids=[1]
+        )
+        self.assertGreater(add_action_result["action_id"], 0)
+
+        actions = cb.watchlist_action_get(watchlist_id=watchlist_id)
+        self.assertIsNotNone(actions)
+        self.assertEqual(len(actions), 1)
+        self.assertEqual(actions[0]["id"], add_action_result["action_id"])
+        self.assertEqual(actions[0]["action_data"], "{\"email_recipients\":[1]}")
+
+        with self.assertRaises(requests.HTTPError) as err:
+            cb.watchlist_action_del(
+                watchlist_id=watchlist_id,
+                action_id=-1
+            )
+        self.assertEqual(err.exception.response.status_code, 500)
+
+        new_actions = cb.watchlist_action_get(watchlist_id=watchlist_id)
+        self.assertIsNotNone(new_actions)
+        self.assertEqual(len(new_actions), 1)
+        self.assertEqual(new_actions[0]["id"], add_action_result["action_id"])
+        self.assertEqual(new_actions[0]["action_data"], "{\"email_recipients\":[1]}")
+        return
+
+    @classmethod
+    def tearDownClass(cls):
+        watchlists = cb.watchlist()
+        del_count = 0
+        for watchlist in watchlists:
+            if watchlist["name"].startswith(name_prefix):
+                cb.watchlist_del(watchlist["id"])
```

Comment:
Might be worth having error handling, so you can continue if one fails... unless it was your intention to stop if one fails